backend/routes/api/seasons.js

```diff
@@ -23,474 +24,490 @@ import {
             },
           ],
         },
+
+        changeLogsQueryPart(),
       ],
     });
 
-    // we want to get all the date types for this season
-    const dateTypes = {};
+    checkSeasonExists(seasonModel);
 
-    // add datetypes from the db to map
-    const dateTypesFromDb = await DateType.findAll({
-      attributes: ["id", "name", "description"],
-      where: {
-        name: {
-          [Op.in]: ["Operation", "Reservation"],
-        },
-      },
-    });
+    // Get the previous year's Season Dates for this Feature
+    const previousSeason = await getPreviousSeasonDates(seasonModel);
 
-    dateTypesFromDb.forEach((dateType) => {
-      dateTypes[dateType.name] = dateType.toJSON();
-    });
+    const output = { current: seasonModel, previous: previousSeason };
+
+    res.json(output);
+  }),
+);
 
-    let features = parkFeatures.map((featureObj) => {
-      const feature = featureObj.toJSON();
+// Get all form data and DateRanges for a Park Season
+router.get(
+  "/park/:seasonId",
+  asyncHandler(async (req, res) => {
+    const seasonId = Number(req.params.seasonId);
 
-      // each feature will have an array of date ranges for the current season and the previous season
-      // we'll remove the dateRanges array and add currentSeasonDates and previousSeasonDates arrays
-      // so that the client can easily sort them
-      feature.dateable.currentSeasonDates = [];
-      feature.dateable.previousSeasonDates = [];
+    const seasonModel = await Season.findByPk(seasonId, {
+      attributes: SEASON_ATTRIBUTES,
+      include: [
+        // Park Area details
+        {
+          model: Park,
+          as: "park",
 
-      feature.dateable.dateRanges = _.orderBy(
-        feature.dateable.dateRanges,
-        "startDate",
-      );
+          include: [
+            // Park-level dates
+            // Dates for this Park Season
+            dateableAndDatesQueryPart(seasonId),
 
-      feature.dateable.dateRanges.forEach((dateRange) => {
-        if (dateRange.seasonId === season.id) {
-          feature.dateable.currentSeasonDates.push(dateRange);
-        }
+            // Park Areas for this Park Season
+            {
+              model: ParkArea,
+              as: "parkAreas",
+              required: false,
+              include: [
+                // Dates for this Park Area Season
+                dateableAndDatesQueryPart(seasonId),
 
-        if (dateRange.seasonId === prevSeason?.id) {
-          feature.dateable.previousSeasonDates.push(dateRange);
-        }
-      });
-      delete feature.dateable.dateRanges;
-      return feature;
-    });
+                {
+                  model: Feature,
+                  as: "features",
+                  include: [
+                    featureTypeQueryPart(),
+
+                    // Dates for this Feature Season
+                    dateableAndDatesQueryPart(seasonId),
+                  ],
+                },
+              ],
+            },
 
```

Comment:
Same as above. Does Park need Park Area and Feature? but we can modify API anytime.